cypress/integration/pages/mediaAssetPage/testsForCanonicalOnly.js

```diff
@@ -41,7 +42,90 @@ export const testsThatFollowSmokeTestConfigForCanonicalOnly = ({
         }
       });
     });
+    describe('Most Watched component', () => {
+      before(() => {
+        cy.getToggles(config[service].name);
+      });
+      // Allows constructing most watched path with variant
+      const serviceVariant = variant === 'default' ? '' : `/${variant}`;
+
+      const mostWatchedPath = `/${config[service].name}/mostwatched${serviceVariant}.json`;
+      it('should show/not show the Most Watched component if the toggle is enabled for the service', function test() {
+        // Find if the service has component enabled in toggles
+        cy.fixture(`toggles/${config[service].name}.json`).then(toggles => {
+          const mostWatchedIsEnabled = path(
+            ['mostPopularMedia', 'enabled'],
+            toggles,
+          );
+          cy.log(`Service toggle enabled ${mostWatchedIsEnabled}`);
+          // If toggle is enabled for that service, also check there are more than 0 records to show
+          if (mostWatchedIsEnabled) {
+            cy.request(mostWatchedPath).then(({ body: mostWatchedJson }) => {
+              if (mostWatchedJson.totalRecords > 0) {
+                cy.log('Enough records to show component');
+                cy.get('[data-e2e=most-watched-heading]').should('exist');
+              } else {
+                cy.log('Not enough records to show component');
+
+                cy.get('[data-e2e=most-watched-heading]').should('not.exist');
+              }
+            });
+          } else {
+            cy.get('[data-e2e=most-watched-heading]').should('not.exist');
+          }
+        });
+      });
+      it('Most Watched component shows correct number of items', function test() {
+        cy.fixture(`toggles/${config[service].name}.json`).then(toggles => {
+          const mostWatchedIsEnabled = path(
+            ['mostPopularMedia', 'enabled'],
+            toggles,
+          );
 
+          cy.request(mostWatchedPath).then(({ body: mostWatchedJson }) => {
+            if (mostWatchedIsEnabled && mostWatchedJson.totalRecords > 0) {
+              const maxNumberofItems = path(
+                ['mostPopularMedia', 'value'],
+                toggles,
+              );
+              cy.log(
+                `Max number of items is ${maxNumberofItems}. Number of items is ${mostWatchedJson.totalRecords}`,
+              );
+              // Compares number of items in json to number of items shown in the component
+              // If there is only one record, the item is not in a list
+              // If the max number of items is 5 and the records are >= 5, checks it shows 5
+              // If the max number of items is 10 and the records are >= 10, checks it shows 10
+              let expectedNumberOfItems;
+              if (mostWatchedJson.totalRecords === 1) {
+                cy.get('[data-e2e=most-watched-heading]').within(() => {
```

Comment:
Do we only want to run this check when there is one record?

Surprised because the check starting line 118 runs whether there's 1 record or not, so curious why we only check for this "wrapper" if there's one record

Reply:
Ah that is a mistake. Line 118 check should only run if more than one record. Will change. I realise this didn't fail because as I was working on this, the example I had with only one item, updated to having 2! The reason for checking for the wrapper when there is only one record is just because I cannot check for it in the same way as I check for the number items in the list, as the list only forms when there is more than 1 item. So for one record I have chosen a different locator for it. This will fail if 1 item is expected but 0 items show, but I think now looking at it again it will not fail if 1 is expected but 2 are shown (not sure how this would be possible to even happen though as there wouldn't be data there).